core/types/transaction_signing.go

```diff
@@ -154,18 +154,30 @@ func (s EIP155Signer) SignatureValues(tx *Transaction, sig []byte) (R, S, V *big
 // Hash returns the hash to be signed by the sender.
 // It does not uniquely identify the transaction.
 func (s EIP155Signer) Hash(tx *Transaction) common.Hash {
-	return rlpHash([]interface{}{
-		tx.data.AccountNonce,
-		tx.data.Price,
-		tx.data.GasLimit,
-		tx.data.FeeCurrency,
-		tx.data.GatewayFeeRecipient,
-		tx.data.GatewayFee,
-		tx.data.Recipient,
-		tx.data.Amount,
-		tx.data.Payload,
-		s.chainId, uint(0), uint(0),
-	})
+	if tx.data.EthCompatible {
+		return rlpHash([]interface{}{
+			tx.data.AccountNonce,
+			tx.data.Price,
+			tx.data.GasLimit,
+			tx.data.Recipient,
+			tx.data.Amount,
+			tx.data.Payload,
+			s.chainId, uint(0), uint(0),
+		})
+	} else {
```

Comment:
golang lint rules would ask you to remove the `else` claus and just return

Reply:
Turns out we don't have golint (or any drop-in replacement) enabled in our golangci config.  If I add it, it gives a ton of errors of different sorts (including some of this rule, e.g. https://github.com/celo-org/celo-blockchain/blob/master/rpc/client.go#L392-L396)

Personally, I am not a fan of this if/else lint rule.  I remember when I first used golint I didn't like it and also didn't like that golint doesn't let you disable individual rules.  But if you like that rule I can change them.